### PR TITLE
[Component with its own class name as service type] LPS-199542 Refactor Helper components from document-library, document-library-opener-google-drive-web, on-demand-admin, portal-workflow and user-associated-data-web modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Previous pull: https://github.com/rafaprax/liferay-portal/pull/1308

Ticket: https://liferay.atlassian.net/browse/LPS-199542

Hey @tinatian I tried to merge the two methods in DLPortletToolbarContributorUtil, but this https://github.com/liferay-core-infra/liferay-portal/pull/5933#issuecomment-1809676521 happened, because the logic of knowing if the show actions is enabled or if a folder does not exists was conflicting.

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:usm

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found.

**Review by rafaprax — APPROVED** [review]

**Comment by tinatian** [comment]
@MrJohn1911 @rafaprax For https://github.com/liferay-core-infra/liferay-portal/pull/5933/commits/bc974135a9ca283af5cac55e51a2b8a90ada6528, please merge the two methods in DLPortletToolbarContributorUtil, as both usages are invoking isShowActionsEnabled first, then getFolder, thus no need to have two methods exposed.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:lima

**Comment by zhangyan0701** [comment]
Issue found:

**The Show Actions option in Document and Media portlet doesn't work.**

Reproduce steps:
- Add a new widget page and add Document and Media portlet to the page.
- Go to Portlet Configuration.
- Enable Show Actions and Save.

Result:
There should be a plus icon show in portlet topper.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:lima

**Comment by MrJohn1911** [comment]
Hey @tinatian I tried to merge the two methods in `DLPortletToolbarContributorUtil`, but this [regression](https://github.com/liferay-core-infra/liferay-portal/pull/5933#issuecomment-1809676521) happened, because the logic of knowing if the show actions is enabled or if a folder does not exists was conflicting.

**Comment by Kvale1733** [comment]
ci:test:usm

**Comment by Kvale1733** [comment]
ci:test:um

**Comment by Kvale1733** [comment]
ci:test:workflow

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/util/DLPortletToolbarContributorUtil.java [hunk 1]
```diff
@@ -22,17 +22,14 @@
...
 
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Reference;
-
 /**
  * @author Roberto Díaz
  */
-@Component(service = DLPortletToolbarContributorHelper.class)
-public class DLPortletToolbarContributorHelper {
+public class DLPortletToolbarContributorUtil {
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/util/DLPortletToolbarContributorUtil.java:28]: Hey @MrJohn1911, transform it in a base class and make its usages extend it. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
Hey @MrJohn1911 , see the above comment, fix and resend. Thx

**Pull request closed without merging**